spark3/src/test/java/org/apache/iceberg/actions/TestRemoveOrphanFilesAction3.java

```diff
@@ -110,4 +112,54 @@ public void testSparkCatalogHiveTable() throws TableAlreadyExistsException, NoSu
         results.contains("file:" + location + "/data/trashfile"));
   }
 
+  @Test
+  public void testSparkSessionCatalogHadoopTable() throws Exception {
+    spark.conf().set("spark.sql.catalog.spark_catalog", "org.apache.iceberg.spark.SparkSessionCatalog");
+    spark.conf().set("spark.sql.catalog.spark_catalog.type", "hadoop");
+    spark.conf().set("spark.sql.catalog.spark_catalog.warehouse", tableLocation);
+    SparkSessionCatalog cat = (SparkSessionCatalog) spark.sessionState().catalogManager().v2SessionCatalog();
+
+    String[] database = {"default"};
+    Identifier id = Identifier.of(database, "table");
+    Map<String, String> options = Maps.newHashMap();
+    Transform[] transforms = {};
+    cat.createTable(id, SparkSchemaUtil.convert(SCHEMA), transforms, options);
+    SparkTable table = (SparkTable) cat.loadTable(id);
+
+    spark.sql("INSERT INTO default.table VALUES (1,1,1)");
+
+    String location = table.table().location().replaceFirst("file:", "");
+    new File(location + "/data/trashfile").createNewFile();
+
+    List<String> results = Actions.forTable(table.table()).removeOrphanFiles()
+        .olderThan(System.currentTimeMillis() + 1000).execute();
+    Assert.assertTrue("trash file should be removed",
+        results.contains("file:" + location + "/data/trashfile"));
+  }
+
+  @Test
+  public void testSparkSessionCatalogHiveTable() throws Exception {
+    spark.conf().set("spark.sql.catalog.spark_catalog", "org.apache.iceberg.spark.SparkSessionCatalog");
+    spark.conf().set("spark.sql.catalog.spark_catalog.type", "hive");
+    SparkSessionCatalog cat = (SparkSessionCatalog) spark.sessionState().catalogManager().v2SessionCatalog();
+
+    String[] database = {"default"};
+    Identifier id = Identifier.of(database, "sessioncattest");
+    Map<String, String> options = Maps.newHashMap();
+    Transform[] transforms = {};
+    cat.dropTable(id);
+    cat.createTable(id, SparkSchemaUtil.convert(SCHEMA), transforms, options);
+    SparkTable table = (SparkTable) cat.loadTable(id);
+
+    spark.sql("INSERT INTO default.sessioncattest VALUES (1,1,1)");
+
+    String location = table.table().location().replaceFirst("file:", "");
+    new File(location + "/data/trashfile").createNewFile();
+
+    List<String> results = Actions.forTable(table.table()).removeOrphanFiles()
+        .olderThan(System.currentTimeMillis() + 1000).execute();
```

Comment:
I think this test case only works because `HiveCatalogs` uses the value of `hive.metastore.uris` from the environment's `hive-site.xml`. By removing `spark_catalog` and then using the `DataFrameReader`, the Hive catalog from `HiveCatalogs` is used, which has the same URI.

I think this is actually the right thing to do, but I would do it more directly and obviously so that it is clear what is happening:

1. Get the session catalog from the catalog manager
2. If the session catalog is a SparkSessionCatalog, get the underlying Iceberg catalog
3. Use the Iceberg catalog to load the metadata table, because it accepts the full table identifier

Reply:
Discussed on slack: Notes here

Main reason I didn't do this originally is we then need to break the code in to a Spark3 and Spark2 versions. But if we could, we wouldn't have to do this at all since we could just in the Spark 3 mode load the table by getting the Catalog directly and loading from there, instead of trying to fall back to the non-catalog path. Thinking we are going to go down the path of attempting to get Spark Version specific code in here.

Reply:
More notes for myself, Once we have direct access to to catalog manager, let's just do 

```java
String[] ns = {"default", "sessioncattest"};
Dataset.ofRows(spark, DataSourceV2Relation.create(cat.loadTable(Identifier.of(ns, "entries")), Option.apply(cat), Option.apply(id)));
```